### PR TITLE
fix app starting offscreen

### LINKED ISSUE #517: Now the app itself is not opening
I submitted an issue saying that it's not downloading, now the app is closed and it's not opening. I tried restarting my laptop but it's not opening.
What do I do?

### PR DESCRIPTION
**Pull request by Araxeus**
Aims to fix issues such as #193 #542, probably fix #517

checks if the app is completely offscreen (x / width) or if the titlebar is offscreen (y / height) 
> checks if window is inside the display closest to the given point

**Tested on multiple monitor setup and works perfectly**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -124,7 +124,14 @@ function createMainWindow() {
 	remote.enable(win.webContents);
 	if (windowPosition) {
 		const { x, y } = windowPosition;
-		win.setPosition(x, y);
+		if(x + win.getSize()[0] < 0 || x - win.getSize()[0] > electron.screen.getDisplayNearestPoint({x, y}).bounds.width) {
```
Comment (th-ch) [inline index.js]: No need to do the same with y/height?

Comment (Araxeus) [reply index.js]: Well usually the error occurs because x value is offscreen but just to make sure ill add it

**Comment by Araxeus** [comment]
now also check if the titlebar y coordinate is offscreen (if its offscreen then there isn't a way to move the app)

**Comment by Araxeus** [comment]
f5e22b5 made it so that the app validate the coordinates if they are inside the coordinates of the closest display to the given point

* get saved absolute coordinates (grid start point (0,0)  is the top left corner of **the primary monitor**)
* get monitor closest to said point
* check if given coordinates is inside the rectangle of the closest monitor

File: index.js [hunk 2]
```diff
@@ -122,9 +122,20 @@ function createMainWindow() {
...
 	});
 	remote.enable(win.webContents);
+
 	if (windowPosition) {
 		const { x, y } = windowPosition;
-		win.setPosition(x, y);
+		const winSize = win.getSize();
+		const displaySize = electron.screen.getDisplayNearestPoint(windowPosition).bounds;
+		if((x + winSize[0] < displaySize.x - 8 || x - winSize[0] > displaySize.x + displaySize.width) ||
+		   (y < displaySize.y - 8 || y > displaySize.y + displaySize.height)) {
```
Comment (th-ch) [inline index.js:131]: Shouldn't it be `y + winSize[1]` similar to `x + winSize[0]`? (+  nit:the condition only contains `or` operators so inner parentheses can be omitted)

**Review by th-ch — COMMENTED** [review]
Looks good, just one question before merging!

File: index.js (hunk 2 above)
Comment (Araxeus) [reply index.js:131]: if y < displaySize.y then the titlebar is out of the screen and the app can't be controlled

y + winSize[1] would refer to the bottom corner, and its an error if the top corner is out of the screen (just y value)

also the parentheses could be indeed omitted, if you want

**Review by th-ch — APPROVED** [review]

**Pull request merged**